pennylane/measurements/expval.py

```diff
@@ -144,6 +144,17 @@ def process_counts(self, counts: dict, wire_order: Wires):
             probs = qml.probs(wires=self.wires).process_counts(counts=counts, wire_order=wire_order)
         return self._calculate_expectation(probs)
 
+    def process_density_matrix(
+        self, density_matrix: Sequence[complex], wire_order: Wires
+    ):  # pylint: disable=unused-argument
+        if not self.wires:
+            return qml.math.squeeze(self.eigvals())
+        with qml.queuing.QueuingManager.stop_recording():
+            prob = qml.probs(wires=self.wires).process_density_matrix(
+                density_matrix=density_matrix, wire_order=wire_order
+            )
+        return self._calculate_expectation(prob)
```

Comment:
I don't think this is mathematically correct if the density matrix is a mixed state?

Reply:
Wait I think this is in fact correct, nvm. 😅

Reply:
Yes should be find as long as what I imagined is correct: here we have the circuit after application of the measurement, via probs we get the pk = [<k|\rho|k>], and we already had the eigvals seemingly so it's vk = <k|O|k>, here k is the eigbases of O. Both the trace and the probability dist of mixed states are summation so they commute, so basically eventually we can still get the correct expval by simply calc the pkvk

Reply:
Though I'm not sure if the whole pipeline of previous pennylane is really doing what I thought here

Reply:
Note: as long as we excute the circuit the states are always automatically rotated into the computational basis so this is totally fine